### DISCUSSION TITLE
Secure connections

### DISCUSSION THREAD

**Discussion by jmgiaever** [zwave-js/zwave-js-ui#481] [Q&A]
Hi,

I'll see that we now can use secure connections (ssl/wss), and I guess this is so it can be exposed to the internet? I'm wondering how other users is handling this, since there's not any form of authentication and it will then be exposed/accesible to anyone?

**Comment by robertsLando** [comment]
you need to use nginx, you can use it even to create a basic authentication

**Reply by jmgiaever** [reply]
Yeah, that was what I had planned. But I saw the SSL options and wanted to ask the usage of SSL on zjs2m, if it's already behind nginx?

**Reply by robertsLando** [reply]
Auth is on my TODOs. Just need to find some time to implement it :) BTW also other users are asking for https, I will add an env var for that

**Comment by cbho** [comment]
Another reason for having SSL (other than exposing it directly to the internet) is integration of zwavejs2mqtt as iframe in Home Assistant:
["If you are accessing Home Assistant over HTTPS using SSL, you cannot access HTTP sites through an iframe panel."](https://www.home-assistant.io/integrations/panel_iframe/)

Don't now if a workaround could be exposing the interface to the internet via NGINX, e.g., and then use the public address in the iframe - but I think keeping_ it unexposed to the internet and only via Home Assistant (not necessarily needing auth per se) would be preferable.

Regards,
Chr.

**Comment by robertsLando** [comment]
PR Ready #535
